lib/shopify-cli/js_deps.rb

```diff
@@ -0,0 +1,82 @@
+require 'shopify_cli'
+
+module ShopifyCli
+  ##
+  # ShopifyCli::JsDeps ensures that all JavaScript dependencies are installed for projects.
+  #
+  class JsDeps
+    include SmartProperties
+
+    property :ctx, accepts: ShopifyCli::Context, required: true
+
+    ##
+    # Proxy to ShopifyCli::JsDeps.new.install.
+    #
+    def self.install(ctx, verbose = false)
+      new(ctx: ctx).install(verbose)
+    end
+
+    ##
+    # Installs all of a project's JavaScript dependencies using Yarn or NPM, based on the project's settings.
+    #
+    # #### Parameters
+    # - `verbose`: whether to run the installation tools in silent mode.
+    #
+    # #### Example
+    #
+    #   ShopifyCli::JsDeps.new(ShopifyCli::Context context).install
+    #
+    def install(verbose = false)
+      CLI::UI::Frame.open(ctx.message('core.js_deps.installing', yarn? ? 'yarn' : 'npm')) do
+        yarn? ? yarn(verbose) : npm(verbose)
+      end
+      ctx.done(ctx.message('core.js_deps.installed'))
+    end
+
+    private
+
+    def yarn?
+      File.exist?(File.join(ctx.root, 'yarn.lock')) && CLI::Kit::System.system('which', 'yarn').success?
+    end
+
+    def yarn(verbose = false)
+      cmd = %w(yarn install)
+      cmd << '--silent' unless verbose
+
+      success = CLI::Kit::System.system(*cmd, chdir: ctx.root) do |out, err|
+        ctx.puts out
+        err.lines.each do |e|
+          ctx.puts e
+        end
+      end.success?
+      return false unless success
```

Comment:
Some of this seems redundant, I don't think you need the intermediate `success` variable.

Reply:
Indeed, it doesn't seem to be needed in the end. The `success?` call at the end returns a boolean and the return gymnastics here basically amount to returning it as a boolean.

Reply:
I missed that this was existing code (GitHub had hidden the deleted file) so treat this as optional.